### DISCUSSION TITLE
Cannot deep filter sub document using strapi.query

### DISCUSSION THREAD

**Discussion by danny-dang** [strapi/strapi#7587] [Legacy Topics]
Hello, I have a `User` collection that is linked to the `user_datum` collection.
In the `user_datum` collection has a `name` field.
I want to search for the entries with the `name` field using `strapi.query`
How do I do it?
I'm using Mongoose
I have tried the following but didn't get any result:
```
strapi.query('user', 'users-permissions')
  .model
  .find({
    'user_datum.name': 'John Doe'
  })
  .populate({
    path: 'user_datum'
  })
```
This doesn't work either:
```
strapi.query('user', 'users-permissions')
  .find({
    user_datum: {
      name: 'John Doe'
    }
  },['user_datum'])
```
I had this error: `CastError: Cast to ObjectId failed for value "{ name: 'John Doe' }" at path "user_datum" for model "user"`

It works when I'm using GraphQL query. However, I don't want to use it because I want to do some logic in the backend, without using GraphQL query.

I'm using Strapi v3.0.1

**Comment by derrickmehaffy** [comment]
Hello @danny-dang the object you send to the find query needs to be in the same style format you would pass over REST query parameters:

![image](https://user-images.githubusercontent.com/8593673/91228294-d80ac700-e6dc-11ea-9602-01bc708f68be.png)

Or in this case something like:

```js
    strapi.query('test').find({
      "relation.lookup_string": "test1"
    }, [ 'populate_whatever_here' ]);
```
